public/app/features/alerting/unified/components/rules/AlertInstanceStateFilter.tsx

```diff
@@ -1,22 +1,34 @@
-import React from 'react';
+import { capitalize } from 'lodash';
+import React, { useMemo } from 'react';
 
-import { RadioButtonGroup, Label } from '@grafana/ui';
-import { GrafanaAlertState } from 'app/types/unified-alerting-dto';
+import { Label, RadioButtonGroup } from '@grafana/ui';
+import { GrafanaAlertState, PromAlertingRuleState } from 'app/types/unified-alerting-dto';
+
+export type InstanceStateFilter = GrafanaAlertState | PromAlertingRuleState.Pending | PromAlertingRuleState.Firing;
 
 interface Props {
   className?: string;
-  stateFilter?: GrafanaAlertState;
-  onStateFilterChange: (value: GrafanaAlertState | undefined) => void;
+  filterType: 'grafana' | 'prometheus';
+  stateFilter?: InstanceStateFilter;
+  onStateFilterChange: (value: InstanceStateFilter | undefined) => void;
 }
 
-export const AlertInstanceStateFilter = ({ className, onStateFilterChange, stateFilter }: Props) => {
-  const stateOptions = Object.values(GrafanaAlertState).map((value) => ({
-    label: value,
-    value,
-  }));
+const grafanaOptions = Object.values(GrafanaAlertState).map((value) => ({
+  label: value,
+  value,
+}));
+
+const promOptionValues = [PromAlertingRuleState.Firing, PromAlertingRuleState.Pending] as const;
```

Comment:
Interesting, TIL about `as const`! But why do we need it here, it seems like if we remove it the compiler complains 🤔

Reply:
Ah I think I get it, the compiler infers `PromAlertingRuleState[]` but that's incompatible with `InstanceStateFilter` since that's a union of _specific_ `PromAlertingRuleState` states (`Pending | Firing`)

Reply:
Without `as const` the type of `promOptionValues` would be `Array<PromAlertingRuleState>`. Adding `as const` cause that the type is a tuple `[PromAlertingRuleState.Firing, PromAlertingRuleState.Pending]` so the `PromAlertingRuleState.Inactive` in not allowed